sdk/scheduler/src/test/java/com/mesosphere/sdk/scheduler/uninstall/UninstallSchedulerTest.java

```diff
@@ -94,41 +115,46 @@ public void testEmptyOffers() throws Exception {
     @Test
     public void testInitialPlan() throws Exception {
         UninstallScheduler uninstallScheduler = getUninstallScheduler();
-        Plan plan = uninstallScheduler.getPlanCoordinator().getPlanManagers().stream().findFirst().get().getPlan();
+        Plan plan = getUninstallPlan(uninstallScheduler);
         // 1 task kill + 3 unique resources + deregister step
         List<Status> expected = Arrays.asList(Status.PENDING, Status.PENDING, Status.PENDING, Status.PENDING, Status.PENDING);
         Assert.assertEquals(plan.toString(), expected, getStepStatuses(plan));
     }
 
     @Test
     public void testInitialPlanTaskResourceOverlap() throws Exception {
-        // Add TASK_B, which overlaps partially with TASK_A.
-        stateStore.storeTasks(Collections.singletonList(TASK_B));
+        // Add TASK_B and TASK_C:
+        // - TASK_B overlaps partially with TASK_A, but is on a different agent so resources aren't deduped.
+        // - TASK_C meanwhile is on the same agent as TASK_A, so its resources ARE deduped.
+        stateStore.storeTasks(Arrays.asList(TASK_B, TASK_C));
 
         UninstallScheduler uninstallScheduler = getUninstallScheduler();
-        Plan plan = uninstallScheduler.getPlanCoordinator().getPlanManagers().stream().findFirst().get().getPlan();
-        // 2 task kills + 4 unique resources + deregister step.
+        Plan plan = getUninstallPlan(uninstallScheduler);
         List<Status> expected = Arrays.asList(
-                Status.PENDING, Status.PENDING, Status.PENDING, Status.PENDING, Status.PENDING,
-                Status.PENDING, Status.PENDING);
+                Status.PENDING, Status.PENDING, Status.PENDING, // 3 task kills
```

Comment:
If all we're doing is checking for `Status.PENDING` what about checking `.size()` and all equal?

Reply:
I see we test other conditions, so it would be nice if it were easier to map what we are expecting to our expectations.

Reply:
Yeah at the moment it's effectively checking that the step statuses all match a provided snapshot